packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePlugin.kt

```diff
@@ -15,37 +15,189 @@
+import com.amplifyframework.core.model.query.QueryOptions
+import com.amplifyframework.core.model.query.predicate.QueryPredicates
+import com.amplifyframework.core.model.temporal.Temporal
+import com.amplifyframework.datastore.AWSDataStorePlugin
+import com.amplifyframework.datastore.DataStoreException
+import com.amplifyframework.datastore.DataStoreItemChange
+import com.amplifyframework.datastore.appsync.SerializedModel
 import io.flutter.embedding.engine.plugins.FlutterPlugin
 import io.flutter.plugin.common.MethodCall
 import io.flutter.plugin.common.MethodChannel
 import io.flutter.plugin.common.MethodChannel.MethodCallHandler
 import io.flutter.plugin.common.MethodChannel.Result
-import io.flutter.plugin.common.PluginRegistry.Registrar
+import java.util.*
+import kotlin.collections.HashMap
 
 /** AmplifyDataStorePlugin */
-class AmplifyDataStorePlugin: FlutterPlugin, MethodCallHandler {
-  /// The MethodChannel that will the communication between Flutter and native Android
-  ///
-  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
-  /// when the Flutter Engine is detached from the Activity
-  private lateinit var channel : MethodChannel
-
-  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
-    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "amplify_datastore")
-    channel.setMethodCallHandler(this)
-  }
-
-  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
-    if (call.method == "getPlatformVersion") {
-      result.success("Android ${android.os.Build.VERSION.RELEASE}")
-    } else {
-      result.notImplemented()
+class AmplifyDataStorePlugin : FlutterPlugin, MethodCallHandler {
+
+    private lateinit var channel: MethodChannel
+    private val LOG = Amplify.Logging.forNamespace("amplify:flutter:datastore")
+
+    override fun onAttachedToEngine(
+            @NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+        channel = MethodChannel(flutterPluginBinding.binaryMessenger,
+                                "com.amazonaws.amplify/datastore")
+        channel.setMethodCallHandler(this)
+        Amplify.addPlugin(AWSDataStorePlugin())
+        LOG.info("Added DataStore plugin")
+    }
+
+    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+        var data: HashMap<String, Any> = HashMap()
+        try {
+            data = checkArguments(call.arguments)
+        } catch (e: Exception) {
+            prepareError(result, e, FlutterDataStoreFailureMessage.CASTING.toString())
+        }
+        when (call.method) {
+            "query" -> onQuery(result, data)
+            "configure" -> onConfigure(result, data)
+        }
+    }
+
+    private fun onConfigure(flutterResult: Result, request: HashMap<String, Any>) {
+        var modelSchemasMap: List<Map<String, Any>>? = null
+        if (request.containsKey("modelSchemas")) {
```

Comment:
can this be a single line like: 

 if (request.containsKey("modelSchemas") && request["modelSchemas"] is List<*>) {

Reply:
Done.